src/lints/pub_function_changed_abi.ron

```diff
@@ -0,0 +1,58 @@
+SemverQuery(
+    id: "pub_function_changed_abi",
+    human_readable_name: "pub function changed ABI",
+    description: "A public function changed its external ABI.",
+    required_update: Major,
+    reference_link: Some("https://doc.rust-lang.org/reference/items/external-blocks.html#abi"),
+    query: r#"
+    {
+        CrateDiff {
+            baseline {
+                item {
+                    ... on Function {
+                        name @output
+                        visibility_limit @filter(op: "=", value: ["$public"]) @output
+
+                        importable_path {
+                            path @output @tag
+                        }
+                        
+                        span_: span @optional {
+                            filename @output
+                            begin_line @output
+                        }
```

Comment:
Probably best if we grab the span info of the _current_ function (instead of the baseline) so we can point to the file+line in the current code where the user should look.